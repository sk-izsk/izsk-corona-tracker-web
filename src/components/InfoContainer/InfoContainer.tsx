import { Avatar, Box, Grid, makeStyles, Typography, useMediaQuery } from '@material-ui/core';
import clsx from 'clsx';
import { format } from 'date-fns';
import React from 'react';
import { Link } from 'react-router-dom';
import { HomeInfoContainer } from '..';
import { CustomTheme, theme } from '../../theme/muiTheme';
import { capitalizeString } from '../../utils';

export interface Information {
  name: string;
  to: string;
  value: number | undefined;
}

export interface InfoContainerProps {
  information: Information[];
  lastUpdate?: string;
  countryName?: string;
  avatarLink?: string;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  mainContainer: {
    display: 'flex',
    flexDirection: 'column',
    height: '100vh',
  },
  gridContainer: {
    display: 'flex',
    height: '100vh',
    flexDirection: 'row',
  },
  gridContainerMobile: {
    flexDirection: 'column',
    marginTop: theme.spacing(10),
  },
  item: {
    display: 'flex',
    flex: 1,
  },
  textContainer: {
    textAlign: 'center',
    fontWeight: 'bold',
    marginBottom: theme.spacing(1),
  },
  header: {
    position: 'relative',
    top: theme.spacing(10),
    color: theme.palette.primary.light,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  link: {
    textDecoration: 'none',
  },
  download: {
    color: theme.palette.primary.light,
    cursor: 'pointer',
    marginLeft: theme.spacing(0.5),
  },
  avatar: {
    width: theme.spacing(20),
    height: theme.spacing(20),
  },
}));

const InfoContainer: React.FC<InfoContainerProps> = ({ information, lastUpdate, countryName, avatarLink }) => {
  const classes = useStyles();
  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('xs'));
  const handleGetSummary = (countryName?: string) => {
    if (countryName) {
      return window.open(`https://covid19.mathdro.id/api/countries/${countryName}/og`, '_blank');
    }
    return window.open(`https://covid19.mathdro.id/api/og`, '_blank');
  };
  const handleLink = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    e.preventDefault();
  };
  return (
    <>
      <Box className={classes.mainContainer}>
        <Typography className={clsx([classes.textContainer, classes.header])} variant='h3'>
          {countryName ? `${capitalizeString(countryName)} wide cases` : `World wide cases`}
          {avatarLink && <Avatar className={classes.avatar} src={avatarLink} alt={avatarLink} />}
        </Typography>
        <Grid className={clsx([classes.gridContainer, isMobile && classes.gridContainerMobile])} container>
          {information.map((info: Information) => {
            return (
              <Grid className={classes.item} key={info.name} item>
                <Link
                  className={clsx([classes.item, classes.link])}
                  onClick={countryName ? handleLink : undefined}
                  to={info.to}
                >
                  <HomeInfoContainer name={info.name} value={info.value} />
                </Link>
              </Grid>
            );
          })}
        </Grid>
        {lastUpdate ? (
          <Typography className={classes.textContainer} variant='h6'>
            As of {lastUpdate !== undefined && format(new Date(lastUpdate as string), 'KK:mm aa EEEE, do MMMM yyyy')}
            <Typography
              onClick={() => handleGetSummary()}
              className={clsx([classes.textContainer, classes.download])}
              variant='h6'
              component='span'
            >
              Get the summary of world wide cases
            </Typography>
          </Typography>
        ) : (
          <Typography
            onClick={() => handleGetSummary(countryName)}
            className={clsx([classes.textContainer, classes.download])}
            variant='h6'
          >
            {`Get a summary of ${capitalizeString(countryName as string)} wide cases`}
          </Typography>
        )}
      </Box>
    </>
  );
};

export { InfoContainer };
