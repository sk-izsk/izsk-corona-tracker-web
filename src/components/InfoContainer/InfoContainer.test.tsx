import { render } from '@testing-library/react';
import React from 'react';
import { RouterContextProvider } from '../../RouterContextProvider';
import { InfoContainer } from './InfoContainer';

const mockData = {
  information: [
    {
      name: 'mockName',
      value: 123,
      to: '/mockRoute',
    },
  ],
  countryName: 'mock country',
  avatarLink: 'mock link',
  type: 'country',
};

test('InfoContainer renders correctly', () => {
  const { asFragment } = render(
    <RouterContextProvider>
      <InfoContainer {...mockData} />
    </RouterContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
