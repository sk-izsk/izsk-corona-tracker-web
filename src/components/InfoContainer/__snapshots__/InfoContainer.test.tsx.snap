// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-9 makeStyles-mainContainer-1"
  >
    <h3
      class="MuiTypography-root makeStyles-textContainer-5 makeStyles-header-6 MuiTypography-h3"
    >
      Mock country wide cases
    </h3>
    <div
      class="MuiGrid-root makeStyles-gridContainer-2 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-item-4 MuiGrid-item"
      >
        <a
          class="makeStyles-item-4 makeStyles-link-7"
          href="/mockRoute"
        >
          <div
            class="MuiBox-root MuiBox-root-17 makeStyles-mainContainer-10"
          >
            <div
              class="MuiBox-root MuiBox-root-18 makeStyles-halfCircle-11 makeStyles-halfCircle-14"
            />
            <h3
              class="MuiTypography-root makeStyles-header-13 makeStyles-header-16 MuiTypography-h3"
            >
              mockName
            </h3>
            <div
              class="MuiBox-root MuiBox-root-19"
            >
              <div
                class="makeStyles-values-12 makeStyles-values-15"
              >
                0
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
    <h6
      class="MuiTypography-root makeStyles-textContainer-5 MuiTypography-h6"
    >
      As of 08:05 PM Monday, 20th July 2020
      <span
        class="MuiTypography-root makeStyles-textContainer-5 makeStyles-download-8 MuiTypography-h6"
      >
        Get the summary of world wide cases
      </span>
    </h6>
  </div>
   
</DocumentFragment>
`;
